Gems/ExternalProfilers/SuperluminalProfiler/Code/Source/CpuProfiler.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <CpuProfiler.h>
+
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <Superluminal/PerformanceAPI.h>
+
+namespace SuperluminalProfiler
+{
+	void CpuProfiler::Init()
+	{
+		AZ::Interface<AZ::Debug::Profiler>::Register(this);
```

Comment:
Given that only one profiler can be active at a time, I might want to assert there, but ideally I should use a mechanism to say that when this gem is enabled none of the other profiling gem can be enabled, I don't know if we have something like that

Reply:
There is, kinda-of. There is a field in gems.json called `provided_unique_service` that you can set to any string, and the scripts to enable gems should check against this. We did this for PhysX4 vs PhysX5 

https://github.com/o3de/o3de/blob/f29613a5c80d8dbc53016bdaaf3849cd6d77f60f/Gems/PhysX/Core/PhysX4/gem.json#L26

https://github.com/o3de/o3de/blob/f29613a5c80d8dbc53016bdaaf3849cd6d77f60f/Gems/PhysX/Core/PhysX5/gem.json#L26